src/sage/rings/polynomial/polynomial_ring.py

```diff
@@ -1448,24 +1466,64 @@ def random_element(self, degree=(-1,2), *args, **kwds):
             else:
                 raise ValueError("No polynomial of degree >= 0 has all coefficients zero")
 
-        # Pick a random degree
-        d = randint(degree[0], degree[1])
+        if degree == (-1, -1):
+            return 0
+
+        if degree[0] == -1:
+            allow_zero = True
+            degree = (0, degree[1])
+        else:
+            allow_zero = False
+
+        while True:
+            # Pick random coefficients
+            coefs = []
```

Comment:
Rather than constantly appending to a list, it is faster to populate the list with `None` and then modify the entries. Since you know exactly how many elements will be in the list, you can do this.